### DISCUSSION TITLE
Is there plans to add wheelchair costing function?

### DISCUSSION THREAD

**Discussion by PhilipNgo** [valhalla/valhalla#3609] [Q&A] [answered]
Hi,

I was looking through the _graph.lua_ file, intending to create my custom lua profile by following the TaxiCost PR (https://github.com/valhalla/valhalla/pull/1756/files) 
I stumbled across the wheelchair tag and nodes, as well as looking at the pedestrian costing function file. Is the wheelchair already implemented but not in the documentation or is there a plan on adding it and it's currently being developed?

**Comment by dnesbitt61** [comment, marked as answer]
Wheelchair is currently a "type" of pedestrian route in the pedestrian costing. It looks like if this type is set in pedestrian costing then it will look at the wheelchair access flags in the Allowed methods. I remember some testing with max grade (as would impact wheelchair routing) but the min/max grades derived from the elevation data is too noisy at this time to use this as a costing consideration. This does not seem to be documented under pedestrian routing options, but it looks like if you set a costing option of transport_type to wheelchair it might work. There may need to be some additional testing / integration work to enable this properly.
